src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -157,5 +158,59 @@ public override string ToString()
         {
             return "Structure ";
         }
+
+        public static StringValue operator +(Structure val1, Structure val2)
+        {
+            return new StringValue(val1.ToString() + val2.ToString());
+        }
+
+        public static object FromPrimitive(object value)
```

Comment:
I think it's totally appropriate to have multiple classes per file provided only one of them is publicly exposed.  If you have a class who's essentially used by one other class and one other class only and is internal to that class's thinking, I don't think it has to be in a separate file.  A helper class needs to be separate when it's something you expect might be passed back to a caller outside the class, or be passed in by a caller outside the class.  

Essentially, the answer to the question, "what file is this ThatNameClass I'm seeing here defined in?", should always be "look for a file called ThatClassName.cs" and not "look for a file called SomeOtherClassName.cs".  It's only when nobody outside the one place it's used locally will ever see the class and need to find its definition that I'd consider it okay to put it in another class's file.